direct/core/src/main/java/cz/o2/proxima/direct/view/LocalCachedPartitionedView.java

```diff
@@ -101,14 +102,22 @@ protected void onCache(StreamElement ingest, boolean overwrite) {
         attrName = ingest.getAttribute();
       }
       final boolean updated;
-      final Pair<Long, Object> oldVal;
+      final Pair<Long, Payload> oldVal;
       synchronized (cache) {
         oldVal = cache.get(ingest.getKey(), attrName, Long.MAX_VALUE);
+        long sequenceId = ingest.hasSequentialId() ? ingest.getSequentialId() : 0L;
```

Comment:
everywhere  before I see `-1L` as default value for non-filled sequentialId, why is here 0? Maybe I'm missing something.

Reply:
The sequential ID is defined to be positive. Both values 0, and -1 are equally "banned". The value of -1 means, "was not added to the StreamElement upon creation" and value of 0 would mean "probably happened by some error during (de)serialization". It just maybe points out where to look for the problem.

Reply:
Ok, I get it, thanks. Mayebe can we add exactly this info into javadoc for `sequentialId`? I think it will be helpful to understand principles. Probably here https://github.com/datadrivencz/proxima-platform/pull/527/files#diff-868bfa56270432f7b77e8893cb337403328db6863969fea3a22e27d0950fdd2fR237